src/App.js

```diff
@@ -1,28 +1,14 @@
-import { useState, useEffect } from "react";
-import "./App.css";
-
-import { AskName, MainTab } from "./components/";
-
-function App() {
-  const [flag, setFlag] = useState(localStorage.getItem(false));
-  // const funct = async () => {
-  //   try {
-  //     const response = await axios.get(
-  //       "https://api.unsplash.com/photos/random?client_id=AgBYk9WdFOk5290DJb2lZReoAvaQrmRjrrbvd3ay4TY&query=nature&orientation=landscape"
-  //     );
-  //     console.log(response.data.urls.raw);
-  //     setImage(response.data.urls.raw);
-  //   } catch (err) {
-  //     console.log(err);
-  //   }
-  // };
-  useEffect(
-    () =>
-      localStorage.getItem("name") !== null ? setFlag(true) : setFlag(false),
-    []
-  );
-
-  return <div className="App">{flag ? <MainTab /> : <AskName />}</div>;
-}
-
-export default App;
+import { useState, useEffect } from "react";
+import "./App.css";
+
+import { AskName, MainTab } from "./components";
+
+function App() {
+  const [flag, setFlag] = useState(false);
+  useEffect(() =>localStorage.getItem("name") !== null ? setFlag(true) : setFlag(false),

```

Comment:
I think, instead of using `useEffect` you can directly do this thing inside the `useState`, like
```
... = useState(locationStorage...);
```

Why am I saying? As, you have passed an empty array as a dependency so anyways this useEffect will run only once, and I guess, you are using the `useEffect` to initialize the `flag` state, that's why I think instead of using `useEffect` you can directly do this inside the `useState`.

I am guessing, you will not see any change in behaviour after doing what I have suggested. You can try.

Reply:
Yep That's working. Thanks for suggestions :)